FirebaseStorageUI/UIImageView+FirebaseStorage.m

```diff
@@ -0,0 +1,146 @@
+                                                                  id<FUIStorageReference>_Nonnull))completion {
+  return [self sd_setImageWithStorageReference:storageRef
+                                   maxImageSize:5e6 // 5 megabytes
+                               placeholderImage:placeholder
+                                     completion:completion];
+}
+
+- (id<FUIDownloadTask>)sd_setImageWithStorageReference:(id<FUIStorageReference>)storageRef
+                                           maxImageSize:(UInt64)size
+                                       placeholderImage:(nullable UIImage *)placeholder
+                                             completion:(void (^)(UIImage *,
+                                                                  NSError *,
+                                                                  SDImageCacheType,
+                                                                  id<FUIStorageReference>))completion{
+  return [self sd_setImageWithStorageReference:storageRef
+                                   maxImageSize:size
+                               placeholderImage:placeholder
+                                          cache:[SDImageCache sharedImageCache]
+                                     completion:completion];
+}
+
+- (id<FUIDownloadTask>)sd_setImageWithStorageReference:(id<FUIStorageReference>)storageRef
+                                           maxImageSize:(UInt64)size
+                                       placeholderImage:(nullable UIImage *)placeholder
+                                                  cache:(nullable id<FUIImageCache>)cache
+                                             completion:(void (^)(UIImage *,
+                                                                  NSError *,
+                                                                  SDImageCacheType,
+                                                                  id<FUIStorageReference>))completion {
+  NSParameterAssert(storageRef != nil);
+
+  // If there's already a download on this UIImageView, cancel it
+  if (self.sd_currentDownload != nil) {
+    [self.sd_currentDownload cancel];
+    self.sd_currentDownload = nil;
+  }
+
+  // Set placeholder image
+  self.image = placeholder;
+
+  // Query cache for image before trying to download
+  NSString *key = storageRef.fullPath;
+  UIImage *cached = nil;
+
+  cached = [cache imageFromMemoryCacheForKey:key];
+  if (cached != nil) {
+    self.image = cached;
+    if (completion != nil) {
+      completion(cached, nil, SDImageCacheTypeMemory, storageRef);
+    }
+    return nil;
+  }
+
+  cached = [cache imageFromDiskCacheForKey:key];
+  if (cached != nil) {
+    self.image = cached;
+    if (completion != nil) {
+      completion(cached, nil, SDImageCacheTypeDisk, storageRef);
+    }
+    return nil;
+  }
+
+  // If nothing was found in cache, download the image from Firebase Storage
```

Comment:
If the code is ever needed elsewhere I will. Even with this stuff here the whole method is like 20 lines